naloge/algorithms_basics_test.go

```diff
@@ -564,4 +564,178 @@ func TestSumFrequentEvenDivByFour(t *testing.T) {
 			}
 		})
 	}
+}
+
+
+
+
+func TestShortestUniqueDigit(t *testing.T) {
+	tests := map[string]struct {
+		input       []string
+		expected    string
+		errExpected error
+	}{
+		"succes1": {
+			input:    []string{"123", "112", "789", "56"},
+			expected: "56",
+		},
+		"no repeating values1": {
+			input:      []string{"abc", "999", "111"},
+			errExpected: fmt.Errorf("no valid string found"),
+		},
+		"another success": {
+			input:       []string{},
+			errExpected: fmt.Errorf("no valid string found"),
+		},
+		"empty slice1": {
+			input:    []string{"9876", "1234", "88"},
+			expected: "1234",
+		},
+		"yet another succes1": {
+			input:     []string{"9", "98", "987"},
+			expected: "9",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			actual, err := ShortestUniqueDigit(test.input)
+			if err != nil {
+				if test.errExpected != nil {
+					assert.ErrorContains(t, test.errExpected, err.Error())
+				} else {
+					assert.NoError(t, err)
+				}
+			} else {
+				assert.Equal(t, test.expected, actual)
+			}
+		})
+	}
+}
+
+
+
+func TestHowManySumToEven(t *testing.T) {
+    tests := map[string]struct {
+        input    []int
+        expected int
+    }{
+        "unhappy path": {
+            input:    []int{1, 3, 5, 2, 7, 9, 11},
+            expected: 0,
+        },
+        "happy path": {
+            input:    []int{1, 3, 2, 5, 7, 2, 9, 11},
+            expected: 3,
+        },
+        "no odd number path": {
+            input:    []int{2, 4, 6},
+            expected: 0,
+        },
+        "one group path": {
+            input:    []int{1, 3, 5, 7},
+            expected: 1,
+        },
+        "empty input": {
+            input:    []int{},
+            expected: 0,
+        },
+    }
+
+    for name, test := range tests {
+        t.Run(name, func(t *testing.T) {
+            actual := HowManySumToEven(test.input)
+
+            assert.Equal(t, test.expected, actual)
+        })
+
+    }
+}
+
+func TestCommonLowercase(t *testing.T) {
+	tests := map[string]struct {
+		input       []string
+		expected    string
+		errExpected error
+	}{
+		"succes1": {
+			input:    []string{"Hello!", "hello", "HELLO.", "Hi"},
+			expected:"hello",
+		},
+		"no repeating values1": {
+			input:      []string{""},
+			errExpected: fmt.Errorf("no input provided"),
+		},
+		"another success": {
+			input:       []string{"Yes?", "yes!", "YES", "no"},
+			expected: "yes",
+		},
+		"empty slice1": {
+			input:    []string{"One", "Two", "Two", "Three."},
+			expected:"two",
+		},
+		"yet another succes1": {
```

Comment:
ce kopiraš, spremeni imena caseov
```suggestion
		"single word result path": {
```